failsafe/circuit_breaker.py

```diff
@@ -134,6 +141,27 @@ def get_name(self):
         return 'open'
 
 
+class _HalfOpenState:
+    """
+    A status class representing the half open state of a CircuitBreaker
+    """
+
+    def __init__(self, circuit_breaker):
+        self.circuit_breaker = circuit_breaker
+
+    def allows_execution(self):
+        return True
```

Comment:
I might be missing something trivial, but how is it going to make sure that only one request is made if this always returns True?

Reply:
Actually you're spot-on. Technically, while the 1st request for a half-open circuit is being resolved, x more requests could go through, which is not ideal - still, the first one finishing will determine the state of the circuit. 

Ensuring that there's one and only one concurrent request for a half-open circuit is something that would add a good deal of complexity. Let's look at this approach as a more smooth transition from open to closed than the current one, but not completely error-burst free.

Reply:
I've added a new parameter that controls how much _open_ is the half open state.